### PR TITLE
yashim/fix: speed up country checks

### PR DESCRIPTION
**Pull request by yashim-deriv**
Changes:
-   ...

## Type of change

-   [x] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/components/hooks/use-client-country.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { useEffect, useState } from 'react'
+
+export const useClientCountry = () => {
+    const [clients_country, setClientsCountry] = useState('')
+
+    useEffect(() => {
+        const fetchClientCountry = async () => {
+            const response = await fetch('https://www.cloudflare.com/cdn-cgi/trace', {
```
Comment (habib-deriv) [inline src/components/hooks/use-client-country.tsx:8]: Let's get this url from env

Comment (yashim-deriv) [reply src/components/hooks/use-client-country.tsx]: Resolved @habib-deriv  thanks 😄

File: src/store/region-context.tsx [hunk 2]
```diff
@@ -55,28 +54,30 @@ export const RegionProvider = ({ children }: RegionProviderProps) => {
...
-        const is_eu_residence = eu_countries.includes(residence)
-        const is_eu_location = is_eu_residence || (!residence && is_eu_country_ip)
-        const is_eu = is_eu_location || isEuDomain()
-        const is_non_eu = !is_eu
-        const is_cpa_plan = cpa_plan_countries.includes(user_ip_country)
-        const is_latam = latam_countries.includes(user_ip_country)
-        const is_row = !is_eu
-        const is_dev = isLocalhost() || isTestlink()
         if (website_status) {
+            const is_eu_country_ip = eu_countries.includes(client_country)
```
Comment (habib-deriv) [inline src/store/region-context.tsx:64]: Can we move these variables out of useEffect?

Comment (yashim-deriv) [reply src/store/region-context.tsx:65]: I moved most of the things into useEffect. Some things doesn't need to be recalculated on each render. Only when website_status is changed

**Review by habib-deriv — APPROVED** [review]
LGTM

**Review by mohsen-deriv — APPROVED** [review]

**Pull request merged**